include/iodbc.h

```
@@ -98,9 +98,15 @@
 #if	!defined(WINDOWS) && !defined(WIN32_SYSTEM)
 #define _UNIX_
 
+#ifndef _POSIX_C_SOURCE
+/* POSIX.1-2008 is required for strdup() */
+#define _POSIX_C_SOURCE 200809L
+#endif
```

`#else` error out if `_POSIX_C_SOURCE < 200809L`?

---

It already errors, that's how I noticed :)

It would be nice to show the error as early as possible, but there's a small risk of breaking things for people with other C libraries. Clearly it works on glibc _without_ the define, otherwise everyone would have noticed. But does it work on glibc with some earlier constant defined? (I don't know). If it does, then turning that into an error would probably be bad, even though anyone defining `_POSIX_C_SOURCE` to something less than 2008 has asked for it.

---

Yeah, I meant an earlier/more explicit error.

I'm on the side of 'fail fast', so I would welcome such 'breakage', but I know not everyone agrees